message/infrastructure/http/subscriber.go

```diff
@@ -110,8 +141,24 @@ func (s *Subscriber) Subscribe(url string) (chan *message.Message, error) {
 
 // StartHTTPServer starts http server.
 // StartHTTPServer must be called after all subscribe function are called.
-func (s *Subscriber) StartHTTPServer() error {
-	return s.server.ListenAndServe()
+func (s *Subscriber) StartHTTPServer() (chan error, error) {
```

Comment:
If we want this method to be non-blocking and guarantee that the listener is ready after it returns, I guess we have to return a `chan error` for the eventual error when `Serve()` ends and `error` for the immediate error when creating the listener fails.

I've had the HTTP test fail earlier, because the subscriber server wouldn't be ready when the publisher tried to connect to it, as `ListenAndServe` provides no way to wait for the listener to be up.

Reply:
i'm not sure that breaking API for tests to pass is an good idea, and previous API was easier (channels are always problems), maybe for tests we would just wait for open port istead?

Reply:
907188dc4d91e3019e927e0f775727c011289c7f